plugin/kubernetes/external.go

```diff
@@ -87,8 +87,30 @@ func (k *Kubernetes) External(state request.Request) ([]msg.Service, int) {
 func (k *Kubernetes) ExternalAddress(state request.Request) []dns.RR {
 	// If CoreDNS is running inside the Kubernetes cluster: k.nsAddrs() will return the external IPs of the services
 	// targeting the CoreDNS Pod.
-	// If CoreDNS is running outside of the Kubernetes cluster: k.nsAddrs() will return the first non-loopback IP
-	// address seen on the local system it is running on. This could be the wrong answer if coredns is using the *bind*
-	// plugin to bind to a different IP address.
+	// If CoreDNS is running outside of the Kubernetes cluster: k.nsAddrs() will return the non-loopback IPs
+	// that CoreDNS is listening on.
 	return k.nsAddrs(true, state.Zone)
 }
+
+// ExternalServices returns all services with external IPs
+func (k *Kubernetes) ExternalServices(zone string) (services []msg.Service) {
+	zonePath := msg.Path(zone, coredns)
```

Comment:
Still wrapping my head around the whole structure, but shouldn't that be:
```suggestion
	zonePath := msg.Path(zone, "coredns")
```

Reply:
Same thing with https://github.com/coredns/coredns/pull/3419/files#diff-7b62412de4a07f4b4ad2d746cd77c878R59.
Or what am I missing here?

Reply:
It's a constant. From kubernetes.go: `const coredns = "c" // used as a fake key prefix in msg.Service`

This is an artifact of the etcd-styled "Backend" integration in the kubernetes plugin, where records are passed to the back-end in etcd msg structs.

Reply:
Makes sense. What tripped me up was the different usage within the kubernetes PR: 
https://github.com/coredns/coredns/pull/3452/files#diff-82592195683029ae911713a3c6a581fdR84

Reply:
Oh - good catch.  [That](https://github.com/coredns/coredns/pull/3452/files#diff-82592195683029ae911713a3c6a581fdR84) looks like a copy-forward bug from the original kubernetes transfer implementation.  I think it's harmless, but should be fixed there to be consistent with the rest of the k8s code.

Reply:
Sounds good.